### ISSUE TITLE
The `ModificationsHistory` table is not ordered

### ISSUE THREAD

**Issue by xgreenx** [FuelLabs/fuel-core#2095]
The ordering of entries is currently broken for the `ModificationsHistory` because it uses LE instead of BE.

<img width="783" alt="image" src="https://github.com/user-attachments/assets/a93bafc0-02d5-41d2-80ff-9d3eaafe51d1">

The fix is simple, but the problem is that it invalidates the old history. We need to migrate from version 1 to version 2 of the historical database and change all keys to be BE.

**Comment by MuhtasimTanmoy** [comment]
On it.

**Comment by acerone85** [comment]
Some questions related to this one: 

* What's the size of the data that needs to be migrated? 
* What do the keys represent? Are collisions likely when changing the encoding from little endian to big endian, or are collisions not a problem (e.g. when the keys are hashes of some object such as blocks or transactions)?
* Are we allowed any downtime?

Depending on the answer, there are two possible solutions: 

Solution 1. - pre-migrate before start

* Iterate over the key-value pairs in the modifications_history column 
* For each key-pair, copy it to a new table `modifications_history_v2` using the right key format. 
* Delete each key-pair from `modifications_history` as soon as it has been copied to `modifications_history_v2` (need to check if iterators over key-pairs allow it). 
* Once the iterator has finished the traversal, proceed with starting the node. 

Pros: 
* All data is migrated before the node is started, we can eventually drop support for the all key encoding
* Idempotent, if the service stops abruptly during the migration we might need to 
* Collision-resistant. Even if we get collisions

Cons:
* If the size of the data to be migrated is large, there might be too much of a downtime
* The rocksDB module will need to be changed to read modifications from a new column, this process might be error prone.

Solution 2.

* Every time we need to fetch a key-value pair, we encode the key with the V2 encoding. If we find a value, then we are done. 
* Otherwise, we read from the same column-family using the V1 encoding for the key, and if we find a value we write it to the same column-family using the V2 encoding. 
* Writes of new data are done directly using the new V2 encoding for the key

Pros: 

* No downtime, 
* Changes should be local to the modifications_history module,

Cons: 
* Not suitable if collisions are a problem e.g. we have one key whose V1 encoding is the same as the V2 encoding for another key, 
* Keys that are never read will never be migrated, we might have to support V1 for a long time

If we do not have collisions and we are happy with supporting V1 key encoding for the foreseeable future, then I'd go with option 2.

**Comment by xgreenx** [comment]
> What's the size of the data that needs to be migrated?

Default state rewind window is 7d, but potentially it can be for the whole testnet=) So it can be a lot of data ~400 GB.

> What do the keys represent?

We want the state rewind feature still to be available for usage during migration process.

> Are we allowed any downtime?

No, we should avoid downtime if it is possible. Plus, we want all noncritical migrations to be seamless for the end user.

The second solution doesn't work since we want to fix the ordering for this table. It is impossible to fix if we not update all values.

**Comment by acerone85** [comment]
Yep, was about to mention that if we need to iterate over key-value pairs then we also need an additional table

**Comment by acerone85** [comment]
Per discussion on slack: Let's go with option 2 (with a table that keeps track of what data has been migrated), plus a job in the background that is responsible of migrating all the key-value pairs.
